web/web/api/v1/group.py

```diff
@@ -0,0 +1,131 @@
+from flask import request, jsonify, Blueprint
+from marshmallow import ValidationError
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm.exc import NoResultFound, MultipleResultsFound
+
+from .response_wrapper import ApiResponseWrapper
+from web.database import db
+from web.models.group import Group, GroupSchema
+
+group_api_bp = Blueprint('group_api_bp', __name__)
+
+@group_api_bp.route('/groups', methods=['GET'])
+def get_group_ids():
+    '''
+    Retrieve all group objects
+    '''
+    arw = ApiResponseWrapper()
+
+    fields_to_filter_on = request.args.getlist('fields')
+
+    if len(fields_to_filter_on) > 0:
+        for field in fields_to_filter_on:
+            if field not in Group.__table__.columns:
+                arw.add_errors({field: 'Invalid Group field'})
+                return arw.to_json(None, 400)
+    else:
+        fields_to_filter_on = None
+
+    groups = Group.query.all()
+    group_schema = GroupSchema(only=fields_to_filter_on)
+    results = group_schema.dump(groups, many=True)
+    
+    return arw.to_json(results)
+
+
+@group_api_bp.route('/group/<string:group_id>', methods=['GET'])
+def show_group_info(group_id):
+    '''
+    Retrieve one group object
+    '''
+    arw = ApiResponseWrapper()
+    group_schema = GroupSchema()
+
+    try:  
+        group = Group.query.filter_by(group_id=group_id).one()
+    
+    except MultipleResultsFound:
+        arw.add_errors({group_id: 'Multiple results found for the given group id.'})
+        return arw.to_json()
```

Comment:
You're returning an error here, but the status code defaults to 200, which identifies success. Let's return a 400 for now.